apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1448,54 +1450,78 @@ func (u *UniterAPI) ReadSettings(ctx context.Context, args params.RelationUnits)
 	if err != nil {
 		return params.SettingsResults{}, errors.Trace(err)
 	}
+	for i, arg := range args.RelationUnits {
+		settings, err := u.readOneUnitSettings(ctx, canAccessUnit, arg)
+		result.Results[i].Error = apiservererrors.ServerError(err)
+		result.Results[i].Settings = settings
+	}
+	return result, nil
+}
 
-	readOneSettings := func(arg params.RelationUnit) (params.Settings, error) {
-		tag, err := names.ParseTag(arg.Unit)
-		if err != nil {
-			return nil, apiservererrors.ErrPerm
-		}
+func (u UniterAPI) readOneUnitSettings(
+	ctx context.Context,
+	canAccessUnit common.AuthFunc,
+	arg params.RelationUnit,
+) (params.Settings, error) {
+	unitTag, err := names.ParseTag(arg.Unit)
+	if err != nil {
+		return nil, apiservererrors.ErrPerm
```

Comment:
```suggestion
		return nil, apiservererrors.ErrBadRequest
```
Maybe?

Reply:
This is what the uniter is currently expecting. I kept the existing error types.